assets/js/custom/ProgramCredits.js

```diff
@@ -1,50 +1,196 @@
+    editCredits.addClass('danger')
+    editCreditsError.show()
+    editCreditsError.text(error.message)
+  }
+
+  function saveCredits () {
     const newCredits = editCredits.val().trim()
     if (newCredits === credits.text().trim()) {
-      editCreditsUI.addClass('d-none')
-      descriptionHeadline.show()
-      descriptionCreditsContainer.show()
-      handleShowMore()
+      closeCreditsEditor()
       return
     }
 
-    const url = Routing.generate('edit_program_credits', { id: programId, new_credits: newCredits }, false)
+    const languageSelected = creditsSelect.value
+    if (languageSelected === '' || languageSelected === 'default') {
+      const url = Routing.generate('edit_program_credits', { id: programId, new_credits: newCredits }, false)
 
-    $.get(url, function (data) {
-      if (parseInt(data.statusCode) === 200) {
-        location.reload()
-      } else if (parseInt(data.statusCode) === 707) {
-        editCredits.addClass('danger')
-        editCreditsError.show()
-        editCreditsError.text(data.message)
-      }
-    })
+      $.get(url, function (data) {
+        if (parseInt(data.statusCode) === 200) {
+          location.reload()
+        } else if (parseInt(data.statusCode) === 707) {
+          editCredits.addClass('danger')
+          editCreditsError.show()
+          editCreditsError.text(data.message)
+        }
+      })
+    } else if (newCredits === '') {
+      customTranslationApi.deleteCustomTranslation(
+        programId,
+        languageSelected,
+        deleteCustomTranslationSuccess,
+        showCreditsError
+      )
+    } else {
+      customTranslationApi.saveCustomTranslation(
+        programId,
+        newCredits,
+        languageSelected,
+        saveCustomTranslationSuccess,
+        showCreditsError
+      )
+    }
+  }
+
+  function keepOrDiscardChangesResult (result) {
+    if (result.isConfirmed) {
+      previousCreditsIndex = creditsSelect.selectedIndex
+      lastSavedCredits = credits.text().trim()
+    } else if (result.isDenied) {
+      previousCreditsIndex = creditsSelect.selectedIndex
+      getCustomTransaltion()
+    } else if (result.isDismissed) {
+      creditsSelect.selectedIndex = previousCreditsIndex
+    }
+  }
+
+  function resetCreditsEditor () {
+    creditsSelect.selectedIndex = 0
+    previousCreditsIndex = 0
+    const defaultCredits = credits.text().trim()
+    editCredits.val(defaultCredits)
+    lastSavedCredits = defaultCredits
+  }
+
+  function closeCreditsEditor () {
+    descriptionCreditsContainer.show()
+    descriptionHeadline.show()
+    editCreditsUI.addClass('d-none')
+    resetCreditsEditor()
+    handleShowMore()
+  }
+
+  editCreditsButton.on('click', () => {
+    descriptionCreditsContainer.hide()
+    descriptionHeadline.hide()
+    editCreditsUI.removeClass('d-none')
+    showMoreToggle.addClass('d-none')
+  })
+
+  closeCreditsEditorButton.on('click', () => {
+    if (editCredits.val().trim() === lastSavedCredits.trim()) {
```

Comment:
We can make this more readable by extracting this condition as a function

```
function isChangesSaved(..) {
return editCredits.val().trim() === lastSavedCredits.trim()
}
```

Reply:
Added helper method in for both descriptions and credits